syntactic-paths/src/main/java/com/palantir/util/syntacticpath/Path.java

```diff
@@ -51,54 +45,49 @@
  */
 public final class Path implements Comparable<Path> {
 
-    public static final Path ROOT_PATH = new Path(ImmutableList.<String>of(), true, true);
+    public static final Path ROOT_PATH = new Path(new ArrayList<String>(), true, true);
     public static final char SEPARATOR_CHAR = '/';
     public static final String SEPARATOR = "/";
     public static final String BACKWARDS_PATH = "..";
 
-    private static final Splitter PATH_SPLITTER = Splitter.on(SEPARATOR_CHAR).omitEmptyStrings();
-    static final Joiner PATH_JOINER = Joiner.on(SEPARATOR_CHAR);
     private static final char[] ILLEGAL_CHARS = new char[] {0};
-    private static final List<String> ILLEGAL_SEGMENTS = ImmutableList.of(".");
-
+    private static final List<String> ILLEGAL_SEGMENTS = new ArrayList<String>() {
```

Comment:
Using a collection allows the use of Collectsion.disjoint(). Although I think I reimplemented an array version for ILLEGAL_CHAR.

Either way, there's some refactoring I can do